config/manager/manager.yaml

```diff
@@ -29,7 +29,25 @@ spec:
         args:
         - --enable-leader-election
         image: controller:latest
+        imagePullPolicy: IfNotPresent
         name: manager
+        env:
+        - name: SYSTEM_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: WEBHOOK_SECRET_NAME
+          value: $(WEBHOOK_SECRET_NAME)
+        - name: VALIDATING_WEBHOOK_CONFIGURATION
```

Comment:
Kustomize and plain Kubernetes manifests are full of this kind of "hardcoding" of names and configuration. Sure, harding things into application code is generally not a good idea, but this is configuration. It is values that configure running code. - This is the place to set them.

The same goes for DRYing things up. The wrong abstractions always make things harder. IMO it's easier to type out `secret-agent-` once at the beginning of each object name then to have to wonder why what you're seeing in the manifests doesn't match what you're seeing in a cluster.

I do like the idea of getting rid of the namePrefix setting. In my mind it would make sense to use namePrefix if people were going to deploy multiple copies of the operator right next to each other in the same cluster and namespace, but I do think there's a use case for that?